modules/Router.js

```diff
@@ -41,12 +41,10 @@ const Router = React.createClass({
   },
 
   getInitialState() {
-    return {
-      location: null,
-      routes: null,
-      params: null,
-      components: null
-    }
+    // Use initial state from renderProps when available, to allow using match
+    // on client side when doing server-side rendering.
+    const { location, routes, params, components } = this.props
```

Comment:
Thank you for this!  I just got async routes working with RC4, had some issues, but you fixed it for me :)

:+1: @taion